packages/ns-api/files/ns.mwan

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python3
+
+#
+# Copyright (C) 2023 Nethesis S.r.l.
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+import json
+import sys
+
+from euci import EUci
+from nethsec import mwan, utils
+
+cmd = sys.argv[1]
+
+if cmd == 'list':
+    print(json.dumps({
+        'index_policies': {},
+        'store_policy': {
+            'name': 'String',
+            'interfaces': 'String'
+        },
+        'delete_policy': {
+            'name': 'String'
+        },
+        'edit_policy': {
+            'name': 'String',
+            'interfaces': 'String'
+        },
+        'index_rules': {},
+        'store_rule': {
+            'name': 'String',
+            'policy': 'String',
+            'protocol': 'String',
+            'source_address': 'String',
+            'source_port': 'String',
+            'destination_address': 'String',
+            'destination_port': 'String',
+        },
+        'order_rules': {
+            'rules': 'String'
+        },
+        'delete_rule': {
+            'name': 'String'
+        },
+        'edit_rule': {
+            'name': 'String',
+            'policy': 'String',
+            'label': 'String',
+            'protocol': 'String',
+            'source_address': 'String',
+            'source_port': 'String',
+            'destination_address': 'String',
+            'destination_port': 'String',
+        },
+    }))
+elif cmd == 'call':
+    action = sys.argv[2]
+    e_uci = EUci()
+    try:
+        if action == 'index_policies':
+            print(json.dumps({'values': mwan.index_policies(e_uci)}))
+        elif action == 'store_policy':
+            data = json.JSONDecoder().decode(sys.stdin.read())
+            mwan.store_policy(e_uci, data['name'], data['interfaces'])
+            print(json.dumps({'message': 'success'}))
+        elif action == 'delete_policy':
+            data = json.JSONDecoder().decode(sys.stdin.read())
+            mwan.delete_policy(e_uci, data['name'])
+            print({'message': 'success'})
+        elif action == 'edit_policy':
+            data = json.JSONDecoder().decode(sys.stdin.read())
+            mwan.edit_policy(e_uci, data['name'], data['label'], data['interfaces'])
+            print(json.dumps({'message': 'success'}))
+        elif action == 'index_rules':
+            print(json.dumps({'values': mwan.index_rules(e_uci)}))
+        elif action == 'store_rule':
+            data = json.JSONDecoder().decode(sys.stdin.read())
+            mwan.store_rule(
+                e_uci,
+                data['name'],
+                data['policy'],
+                data['protocol'] if 'protocol' in data else None,
+                data['source_address'] if 'source_address' in data else None,
+                data['source_port'] if 'source_port' in data else None,
+                data['destination_address'] if 'destination_address' in data else None,
+                data['destination_port'] if 'destination_port' in data else None,
```

Comment:
These checks should not be needed because:
- all parameters are mandatory
- the same checks are present inside the called function https://github.com/NethServer/python3-nethsec/pull/7/files#diff-400c6e845848e5871253b006dfac1bb3b0c0a005d63b73e45c41e7ca4ca12833R142-R151

Reply:
Mb, thought I've removed the checks even on the store_rule